Extension/src/LanguageServer/extension.ts

```diff
@@ -25,9 +31,11 @@ let ui: UI;
 let disposables: vscode.Disposable[] = [];
 let languageConfigurations: vscode.Disposable[] = [];
 let intervalTimer: NodeJS.Timer;
+let insiderUpdateTimer: NodeJS.Timer;
 let realActivationOccurred: boolean = false;
 let tempCommands: vscode.Disposable[] = [];
 let activatedPreviously: PersistentWorkspaceState<boolean>;
+const insiderUpdateTimerInterval: number = 1000 * 15;
```

Comment:
Why do you need a special timer? I thought we were going to piggyback on the cpptools.json -- i.e. we would just check for a new version whenever we check that (i.e. on startup or every 30 minutes or whatever).

Reply:
We should only check for an update when the user has the `updateChannel` setting set to `Insiders`, I suppose we could instead query what the setting is turned to once the `intervalTimer` goes off. In fact, that might even be the correct thing to do. What do you think?

Reply:
I think piggybacking on the abTesting update timer is fine, but the setInterval call will need to be factored out of the ABTestSettings class.  Whichever way you go, 15 seconds is too frequent to check (though I think you've only set it this way for testing purposes).

Reply:
Yes, 15 seconds is much too often, even for testing! I managed to hit GitHub's 60 requests per hour (for unauthorized accesses) just by leaving debug on for a few minutes. Woops.
At any rate, I think it might be useful to keep this timer independent so as to allow for a configuration point on Insiders updating specifically